Mathlib/Analysis/Normed/Field/Ultra.lean

```diff
@@ -0,0 +1,63 @@
+/-
+Copyright (c) 2024 Yakov Pechersky. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yakov Pechersky
+-/
+import Mathlib.Analysis.Normed.Field.Basic
+import Mathlib.Analysis.Normed.Group.Ultra
+
+/-!
+# Ultrametric norms on division rings
+
+This file contains results on the behavior of norms in ultrametric division rings.
+
+## Main results
+
+* `norm_intCast_le_one`:
+  the norm of the image of an integer in the division ring is always less than or equal to one
+
+## Tags
+
+ultrametric, nonarchimedean
+-/
+open Metric NNReal
+
+namespace IsUltrametricDist
+
+section DivisionRing
+
+variable {K : Type*} [NormedDivisionRing K] [IsUltrametricDist K]
```

Comment:
```suggestion
variable {K : Type*}  [SeminormedRing K] [NormOneClass K] [IsUltrametricDist K]
```
All the lemmas work fine in this more general context (meaning they apply to `ℤ_[p]`). This will of course mean renaming the file, and moving it to `Analysis/Normed/Ring`.